theforeman.org/pipelines/release/source/smart-proxy.groovy

```diff
@@ -8,29 +8,37 @@ pipeline {
         buildDiscarder(logRotator(daysToKeepStr: '7'))
     }
 
-    stages {
-        stage("Clone repository") {
-            steps {
-                git url: git_url, branch: git_ref
+    matrix {
```

Comment:
This is in the error output:
```
2023-03-24T16:20:03.7780551Z {'error': ['Undefined section "matrix" @ line 21, column 5.', 'Missing required section "stages" @ line 11, column 1.']}
```

I think you need this kind of structure:

* stages
  * stage: 'test'
    * matrix
    * stages
      * stage: clone
      * stage: Test Ruby
  * stage: Build and Archive Source

Reply:
@ekohl Thanks for the detailed explaination, it helped! Also locally it is passing all checks and this is the output i got:
`theforeman.org/output/smart-proxy-develop-source-release/config.xml: OK`